### PR TITLE
Warden must be reset after each example with it

### PR DESCRIPTION
**Pull request by cmrd-senya**
see https://github.com/hassox/warden/wiki/Testing#requirements

### CODE REVIEW

**Comment by jhass** [comment]
Can't we do away with this and use only devise's helpers? https://github.com/plataformatec/devise/blob/master/lib/devise/test/controller_helpers.rb

File: spec/integration/mentioning_spec.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def users_connected?(user1, user2)
 
 describe "mentioning", type: :request do
   include MentioningSpecHelpers
+  include RequestsHelper
```
Comment (SuperTux88) [inline spec/integration/mentioning_spec.rb]: Is this the only place where this is used? Why do we need a Helper then, when it is only used once?

**Comment by jhass** [comment]
Porting everything to https://github.com/plataformatec/devise#test-helpers would be great I think.

**Comment by cmrd-senya** [comment]
@jhass, these links are for devise master branch. We use 3.5.6 and it doesnt have `Devise::Test::IntegrationHelpers`. Instead, the documentation [says](http://www.rubydoc.info/github/plataformatec/devise/#Test_helpers):

>These helpers are not going to work for integration tests driven by Capybara or Webrat. They are meant to be used with functional tests only. It is undesirable even to include Devise::TestHelpers during integration tests. Instead, fill in the form or explicitly set the user in session;

**Comment by jhass** [comment]
4.2 has them too, https://github.com/plataformatec/devise/tree/245b1f9de0b3386b7913e14b60ea24f43b77feb0#test-helpers

We really need to upgrade and it looks like the blocker is gone since a couple weeks ago! https://github.com/baschtl/devise-token_authenticatable/releases Mind giving updating it a shot?

**Comment by cmrd-senya** [comment]
> Mind giving updating it a shot?

Not at all :)

**Comment by cmrd-senya** [comment]
Superseded by #6959.

**Pull request closed without merging**